plugins/unicodelib/unicodelib.cpp

```diff
@@ -718,6 +719,43 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+UnicodeString excludeNthWord(RuleBasedBreakIterator& bi, unsigned textLen, UChar const * text, unsigned n)
```

Comment:
This is an improvement, but the return is likely to involve an extra clone of the string - unless the c++ compiler is very clever.

You could do the following instead:

```
void excludeNthWord(RuleBasedBreakIterator& bi, UnicodeString & source, unsigned n)
...
                source.removeBetween(wordBeginning, wordEnd);
                return;
...
    UnicodeString processed(text, textLen);
    excludeNthWord(*bi, processed, n);

```